src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,140 @@
+package duke;
+import java.util.ArrayList;
+
+public class Ui {
+
+    /**
+     * Returns the welcome message of the duke chat bot.
+     * 
+     * @return Welcome Message
+     */
+    public static String WelcomeMessage() {
+        String separator = "    ____________________________________________________________";
+        return separator + "\n" + "     Hello! I'm Duke" + "\n" + "     What can I do for you?"  + "\n" + separator;
+    }
+
+    /**
+     * Returns updated user task list and prints feedback to the user based on their input
+     * 
+     * @param taskList list of user's tasks
+     * @param input user input
+     * @return updated task list
+     */
+    public ArrayList<Task> run(ArrayList<Task> taskList, String input) {
+        ArrayList<Task> list = taskList;
+        String separator = "    ____________________________________________________________";
+        try {
+            String parsedInput = Parser.parse(input);
+            if (parsedInput.equals("list")) {
+                if (list.size() == 0) {
+                    System.out.println(separator);
+                    System.out.println("     You have no tasks in your list!");
+                } else {
+                    System.out.println(separator);
+                    System.out.println("     Here are the tasks in your list:");
+                    for (int i = 0; i < list.size(); i++) {
+                        Task item = list.get(i);
+                        System.out.println(String.format("     %d. %s", (i + 1), item.toString()));
+                    }
+                }
+            } else if (parsedInput.equals("done")) {
```

Comment:
perhaps the throwing of exceptions should be done in the Parser rather than Ui handler, since the parser handles valid and invalid input